docs/build.md

```diff
@@ -248,6 +250,35 @@ spec:
       name: icr-knbuild
 ```
 
+### Sources
+
+Represents remote artifacts, as in external entities that will be added to the build context before the actual build starts. Therefore, you may employ `.spec.sources` to download artifacts from external repositories.
+
+```yaml
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: nodejs-ex
+spec:
+  sources:
+    - name: project-logo
+      type: http
+      url: https://gist.github.com/project/image.png
+      http:
+        path: assets/images/logo.png
+```
+
+Under `.spec.sources` we have the following attributes:
+
+- `.name`: represents the name of resource, required attribute.
+- `.type`: data source type, currently only `http` type is supported, required attribute.
+- `.url`: universal resource location (URL), required attribute.
+- `.http.path`: alternative path to save URL contents, can be either a full or relative path, optional attribute.
+
+When downloading artifacts the process is executed in the same directory where the application source-code is located, by default `/workspace/source`, therefore when using relative path on `.http.path` mind the starting location.
+
+Additionally, we have plan to keep evolving `.spec.sources` by adding more supported types, this API attribute works as an extension point to support external and internal resource locations.
+
```

Comment:
Per https://github.com/shipwright-io/build/pull/616#discussion_r598699056 @otaviof I think you need to put the disclaimer about not supporting auth in the URL here in this doc.

Reply:
Even with the latest version of commits @otaviof I don't see the disclaimer in docs/build.md that the current implementation does not support URL's with auth credentials included.

Or did you change the code to support that?